### ISSUE TITLE
database + honeypot problem : Can't set Field of ContentType

### ISSUE THREAD

**Issue by DamShaa** [bobdenotter/forms#67]
When "database" and "honeypot" are both activated, the form tries to insert the virtual field created by HoneyPot into the ContentType.

"leads" is my ContentType :
Can't set Field 'fax_name_state_company' of 'leads'. Make sure the Field is defined in the Leads ContentType

- Bolt version: 4.1.17
- Install type: Packaged distribution
- Database: sqlite 3.26.0 (without JSON)
- PHP version: 7.4.16
- Symfony version: v5.2.6
- Operating System: Linux - 3.10.0-962.3.2.lve1.5.36.el7.x86_64
- Bolt Forms version: 1.4.1

**Comment by benr77** [comment]
This is also reported in #35 but AFAIK there is currently no solution except to disable the honeypot function.

I have enabled a reCAPTCHA field instead which protects the form enough without the honeypot.

**Comment by ghost** [comment]
Issue confirmed today using Bolt 4.1.19 and Forms 1.4.1.

I'm using the "disable honeypot" workaround for now but this needs a solution because captchas are undesirable (they all have accessibility issues and reCAPTCHA tracks visitors).

**Comment by bobdenotter** [comment]
Correct, this is still a problem we should fix!

**Comment by benr77** [comment]
Why is there a requirement that the honeypot field name must change every time I deploy? I appreciate the field name needs to appear like a real field, but once this is set it should be able to be constant for the rest of the lifetime of the app.

Can we generate it based on the Symfony APP_SECRET, or just allow the user to configure the name directly? Any thoughts?

**Comment by bobdenotter** [comment]
@benr77 It is based on the secret:

```
$seed = preg_replace('/[^0-9]/', '', md5($_SERVER['APP_SECRET'] . $_SERVER['REMOTE_ADDR']));
mt_srand($seed % PHP_INT_MAX);
```

But, wouldn't it changing occasionally be irrelevant, if this bug is fixed? i.e. then it'll change, but it should cause no side-effects, so no harm done, right?

**Comment by benr77** [comment]
Thanks @bobdenotter - I don't see a problem with the field name changing occasionally (e.g. if the secret is changed). 

This would assume that the fix is going to automatically ignore whatever the honeypot field is called when submitting the form to the Content Type.

**Comment by I-Valchev** [comment]
Would it make sense to make a `HoneypotType` field type, then it can fully use all nice things that come with a field, and also it'll be able to ignore a field of type Honeypot in saving to Content Types, sending it off to forms, etc.?

**Comment by bobdenotter** [comment]
I've submitted a PR that fixes this by virtue of adding a new option: #69 

I've also expanded on the docs for this feature: https://github.com/bolt/forms/blob/6d5fafa27bd48bebdc064887a31f213689722a94/docs/handlers/save_to_contenttype.md
